src/com/lftechnology/java/training/alina/jdbc/dao/user/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+
+import com.lftechnology.java.training.alina.jdbc.constants.Constants;
+import com.lftechnology.java.training.alina.jdbc.dao.employee.impl.EmployeeDaoImpl;
+import com.lftechnology.java.training.alina.jdbc.dao.user.UserDao;
+import com.lftechnology.java.training.alina.jdbc.dbutils.DbFacade;
+import com.lftechnology.java.training.alina.jdbc.domain.Database;
+import com.lftechnology.java.training.alina.jdbc.domain.Employee;
+import com.lftechnology.java.training.alina.jdbc.domain.User;
+import com.lftechnology.java.training.alina.jdbc.service.DateTimeService;
+import com.lftechnology.java.training.alina.jdbc.service.UserService;
+import com.lftechnology.java.training.alina.jdbc.views.LoginView;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+
+public class UserDaoImpl implements UserDao {
+
+    private static final Logger LOGGER = Logger.getLogger(UserDaoImpl.class.getName());
+
+    /**
+     * Checks whether employee username/password matches. Logins to the system if password matches.
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @return loginStatus {@link Boolean} login status
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     * @throws SQLException
+     */
+    public boolean checkEmployeeLogin(Scanner scanner) throws SQLException {
+        User user = new User();
+        LoginView.displayLoginHeader();
+        user = UserService.setLoginInfo(scanner, Constants.USER_LOGIN);
+        Boolean loginStatus = false;
+        Map<Integer, Object> params = new HashMap<Integer, Object>();
+        params.put(1, user.getUsername());
+        params.put(2, user.getPassword());
+        params.put(3, Constants.NOT_TERMINATED);
+        params.put(4, Constants.NOT_DELETED);
+        Database database = UserService.setDatabaseParams(params);
+        String sql =
+                "SELECT * FROM user u inner join employee e on (u.user_id = e.user_id) where u.username=? and u.password=? and u.is_terminated=? and e.is_deleted=?";
+        Connection connection = DbFacade.getDbConnection();
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement = DbFacade.setParameterizedObjects(database, preparedStatement);
+        ResultSet result = preparedStatement.executeQuery();
+        if (result.next()) {
+            loginStatus = true;
+            checkValidUser(scanner, result.getInt("user_id"), connection);
+        } else {
+            loginStatus = false;
+        }
+        return loginStatus;
+    }
+
+    /**
+     * Checks if user is valid
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param userId
+     *            {@link Integer}
+     * @param connection
+     *            {@link Connection}
+     * @throws SQLException
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     */
+    private void checkValidUser(Scanner scanner, int userId, Connection connection) throws SQLException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused method parameter(s) "connection". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1172)